### ISSUE TITLE
Questions about TRUST4

### ISSUE THREAD

**Issue by silvia1234567890** [liulab-dfci/TRUST4#203]
Hello, 
I have a few questions regarding TRUST4 as I think it could be very useful for me. 
Could I use TRUST4 to analyze sequences of species that are not in the IMGT? What tool do you use for this? I have the sequences in fasta format, and I know which are the most and least abundant, but I would need to know in those particular sequences which V, which D and which J are expressed. And I have the data from the genome. 

And another question, regarding the alignment, how does TRUST4 align the sequences that we give it and the database?

Thank you very much in advance,
Regards,
Silvia

**Comment by mourisl** [comment]
Yes, you can use sequences not in the IMGT. TRUST4 conducts the alignment itself. Does each sequence a VDJ recombined, or it is still V, D, J-separated as IMGT? TRUST4 assumes each reference sequence to have gene name such as "TRBV", "IGHJ" to know the chain and the VDJ information.

To run TRUST4 on non-IMGT-format sequence, you just put your reference sequence as the argument for the -f option, and DON'T use the --ref option. TRUST4 will use the amino acid motif to determine CDR3 sequence.

**Comment by silvia1234567890** [comment]
Thank you so much for your quick response.

I have TCRalpha sequences. And we have the data from the genome, i.e. we know how many Valfa there are, Jalfa there are,... and we know exactly the sequence. 
Regarding this "Is each sequence a recombinant VDJ, or is it still V, D, J-separated as IMGT?" which commands would I have to use in each case? 
How can I generate my own VDJ database?
The --ref command is only used when you have the IMGT database?

Thank you for your help,
Kind regards,
Silvia

**Comment by mourisl** [comment]
You can just put the sequences in a plain fasta file, and name the genes with the convention (TRAVXX, TRADXX, TRAJXX). This file will be the input for the "-f" option. You are right about the --ref option, and I think you don't need it in your case.

**Comment by silvia1234567890** [comment]
And if my sequences contain UMI information, what kind of information can I extract with TRUST4 and which commands?
Thanks

**Comment by mourisl** [comment]
You need to use "--UMI" option to specify the read file containing the UMI sequence. Then you may use the command "--readFormat XXX" to specify the read structure, where the "um" keyword can describe the range of UMI on the read. If there is little conflict between the UMIs from different chains, you can use the option "--barcodeLevel molecule" to assemble the sequence for each UMI separately and then get the abundance with UMI count.

**Comment by silvia1234567890** [comment]
Hi,

just to confirm I'm doing well. I would like to use TRUST4 to analyze sequences of species that are not in the IMGT, and I have the sequences in fasta format (with V, D, J-separated as IMGT). And I need to know in these concrete sequences which V, which D and which J are expressed. 

As input files I would use this fasta file with "-1/-2" option (for paired-end). And I should generate my own VDJ database (a fasta file with the genes following the convention TRAVXX, TRADXX, TRAJXX).  And this file would be the input for the "-f" option. And since the sequences are not in IMGT database I don't need to use the "--ref" option. 

The command line would be:
./run-trust4 -1 read1.fastq -2 read2.fastq -f tcr.fasta

Is everything OK?

Thank you in advance.
Kind regards,
Silvia

**Comment by mourisl** [comment]
Yes, this looks right. You can use options like "-t XXX" for better running speed.
